### PR TITLE
[SYCLomatic] Refine the cufft API migration using new helper functions

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/test/dpct/helper_files_ref/include/fft_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,936 @@
...
+  complex_float_to_complex_float,
+  complex_double_to_complex_double,
+};
+
+/// A class to perform FFT calculation.
+class fft_engine {
+public:
+  /// Default constructor.
+  fft_engine() {}
+  /// Commit the config for calculate n-D FFT.
```
Comment (YuriPlyakhin) [inline clang/test/dpct/helper_files_ref/include/fft_utils.hpp]:
```suggestion
  /// Commit the configuration to calculate n-D FFT.
```

Comment (zhiweij1) [reply clang/test/dpct/helper_files_ref/include/fft_utils.hpp]: Fixed

**Review by tomflinda — APPROVED** [review]
LGTM

**Comment by zhiweij1** [comment]
The failed libcu_atomic case on windows gpu is a known issue.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**